### PR TITLE
Set default ns with `FLUX_SYSTEM_NAMESPACE` env var

### PR DESCRIPTION
**Pull request by stefanprodan**
This PR allows users to change the default namespace from `flux-system` to some other value using the `FLUX_SYSTEM_NAMESPACE` environment variable. 

Ref: https://github.com/fluxcd/flux2/discussions/1284

### CODE REVIEW

**Comment by hiddeco** [comment]
How can it be that your example contains `gitops-system` default in the doc note?

```
  -n, --namespace string    the namespace scope for this operation, can be set with FLUX_SYSTEM_NAMESPACE env var (default "gitops-system")
```

File: cmd/flux/main.go [hunk 1]
```diff
@@ -152,6 +153,16 @@ func configureKubeconfig() {
...
 
+func getDefaultNamespace() string {
+	fromEnv := os.Getenv("FLUX_SYSTEM_NAMESPACE")
+	switch {
+	case fromEnv != "":
+		return fromEnv
+	default:
+		return rootArgs.defaults.Namespace
+	}
+}
```
Comment (hiddeco) [inline cmd/flux/main.go:164]: The logged doc default should not take the set env into account I think.

Comment (stefanprodan) [reply cmd/flux/main.go:164]: I think it should, people should be aware they've changed the default, `--help` would display that.

**Comment by stefanprodan** [comment]
> How can it be that your example contains gitops-system as default in the doc note?

This is what the current PR does, it allows people to override the **default value** of `--namespace` with an env var.

File: cmd/flux/main.go (hunk 1 above)
Comment (hiddeco) [reply cmd/flux/main.go:164]: But that no longer makes it display the default, but rather the current configuration. The default is still `flux-system` if the environment variable is `unset`, which is the information I would like to get from documentation.

Now I no longer know what it falls back to if I would `unset`, wile I could still do an `env | grep FLUX_SYSTEM_NAMESPACE` after having read "can be set with [..]".

Comment (stefanprodan) [reply cmd/flux/main.go:164]: > Now I no longer know what it falls back to if I would unset it

```go
func getDefaultNamespace() string {
	fromEnv := os.Getenv("FLUX_SYSTEM_NAMESPACE")
	switch {
	case fromEnv != "":
		return fromEnv
	default:
		return rootArgs.defaults.Namespace
	}
}
```

it goes to `default` in the switch.

Comment (hiddeco) [reply cmd/flux/main.go:164]: But that still requires you to unset the environment variable, which is what I describe above. I am going to need a second opinion on this.

Comment (stefanprodan) [reply cmd/flux/main.go:164]: This works the same as the `KUBECONFIG` env var for kubectl, if found in env it uses that as the default, if you set the arg to some other value it uses that.

Comment (hiddeco) [reply cmd/flux/main.go:164]: Yes, that is how it should work technically which is fine. The point I am not sure about is making the documentation change based on an environment variable that is set, I would expect it to be static.

**Review by hiddeco — APPROVED** [review]

**Pull request merged**